src/dfx/src/lib/builders/assets.rs

```diff
@@ -126,13 +126,19 @@ impl CanisterBuilder for AssetsBuilder {
             })
             .collect::<DfxResult<Vec<CanisterId>>>().with_context( || format!("Failed to collect dependencies (canister ids) of canister {}.", info.get_name()))?;
 
-        let vars = super::environment_variables(info, &config.network_name, pool, &dependencies);
+        let vars = super::get_and_write_environment_variables(
+            info,
+            &config.network_name,
+            pool,
+            &dependencies,
+            config.env_file.as_deref(),
+        );
```

Comment:
Just a little surprised the error is handled at the subsequent call site, rather than here
```suggestion
        )?;
```